src/main/perl/Path.pm

```diff
@@ -214,17 +216,20 @@ sub _safe_eval
     %opts = %$optsref if $optsref;
 
     local $@;
-    eval {
-        $res = $funcref->(@args, %opts);
-    };
+    my $status = eval {
+                       $res = $funcref->(@args, %opts);
+                      };
 
-    if ($@) {
-        chomp($@);
-        return $self->fail("$failmsg: $@");
-    } else {
-        my $res_txt = defined($res) ? "$res" : '<undef>';
-        chomp($res);
+    # $status is undef if there is a syntax or runtime error or if $res is undef.
+    if ( defined($status) ) {
```

Comment:
I think this is the crucial point. `@_` is true only if there was a syntax or runtime error. But if `$res` is `undef` when the evaluated function completed but returned `undef` because of an error, then this is not catched by testing `@_ `. It took me quite some time to understand that is was the reason one of the test was failing.